skycatalogs/objects/snana_object.py

```diff
@@ -38,12 +36,22 @@ def get_gsobject_components(self, gsparams=None, rng=None):
         return {'this_object': galsim.DeltaFunction(gsparams=gsparams)}
 
     def get_observer_sed_component(self, component, mjd=None):
+        if mjd is None:
+            mjd = self._belongs_to._mjd
+        if mjd is None:
+            txt = 'SnananObject._get_sed: no mjd specified for this call\n'
```

Comment:
`'SnananObject'` -> `'SnanaObject'`

I still object to this test being here and to not having an `SnanaCollection.__init__` that tests if `mjd` is `None`.  At the very least, that `__init__` function should emit a warning in this case.